helper/error-handler.ts

```diff
@@ -1,80 +1,81 @@
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Internal Server Error'}`;
-    case 502:
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Bad Gateway'}`;
-    case 503:
-      throw `[500] ${error.name ? `${error.name}. ` : ''}${error.message || 'Service Unavailable'}`;
-    case 504:
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Gateway Timeout'}`;
-    default:
-      throw `[500] ${error.name ? `${error.name}. ` : ''}${error.message || 'Internal Server Error'}`;
-  }
-}
+import { HttpBadRequestError, HttpInternalServerError } from '@errors/http';
+import { ErrorStatusCode, HttpError } from '@errors/http/http-error';
+import { InputValidationError } from '@errors/runtime';
+import { RuntimeError } from '@errors/runtime/runtime-error';
+import { log } from '@helper/logger';
+import { format } from '@redtea/format-axios-error';
+import { AxiosError } from 'axios';
+
+export function errorHandler(caughtError: Error | HttpError | AxiosError | RuntimeError): undefined {
+  let error = caughtError;
+  const axiosError = (caughtError as AxiosError).isAxiosError && format(caughtError as AxiosError);
+
+  if (!(error instanceof HttpError)) {
+    /**
+     * It means that error was unexpected and can have unpredictable structure
+     * For example, Axios errors have different structure
+     * Also, we can send all unexpected errors to Sentry here
+     */
+    log('This error was not generated by us. We should extract the statusCode, name and message here');
+
+    error = formatUnknownError(error, axiosError);
+  }
+
+  /**
+   * console.log allows displaying the error message and stack trace
+   * But Axios error has complicated structure that doesn't allow debugging it easily
+   */
+  if (axiosError) {
+    console.log(axiosError);
+  } else {
+    console.log(caughtError);
+  }
+
+  /**
+   * Serverless supports following status codes:
+   * 400  Bad Request
+   * 401  Unauthorized
+   * 403  Forbidden
+   * 404  Not Found
+   * 422  Unprocessable Entity
+   * 500  Internal Server Error
+   * 502  Bad Gateway
+   * 504  Gateway Timeout
+   */
+  error = formatUnsupportedError(error as HttpError);
+
+  /**
+   * The error message looks like: [404] Not Found. User does not exist
+   */
+  throw `[${(error as HttpError).statusCode}] ${error.name}. ${error.message}`;
+}
+
+function formatUnknownError(error: Error | AxiosError | RuntimeError, axiosError): HttpError {
+  if (error instanceof InputValidationError) {
+    return new HttpBadRequestError(error.message);
+  }
+  if (error instanceof RuntimeError) {
+    return new HttpInternalServerError(error.message);
+  }
+
+  if (axiosError) {
+    return {
+      statusCode: axiosError.response?.status as ErrorStatusCode,
+      message: axiosError.response?.statusText || axiosError.message,
+      name: axiosError.response?.statusText || 'Internal Server Error',

```

Comment:
Why don't you create proper HttpError instances here?

Reply:
axios can return different errors